src/memory_mapped_file.hpp

```diff
@@ -4,34 +4,15 @@
 #include <cstring>
 #include <string>
 #include <string_view>
-
-#ifdef _WIN32
```

Comment:
why are we making such a huge change to `memory_mapped_file`?
are we assuming this will only get run on Windows?